### PR TITLE
Overlays: deprecate closeOnOutsideClick and publish hideOnOutsideClick

### PR DESCRIPTION
**Pull request by ksercs**

### CODE REVIEW

**Review by Zedwag — DISMISSED** [review]

File: js/ui/context_menu/ui.context_menu.js [hunk 1]
```diff
@@ -475,7 +482,7 @@ class ContextMenu extends MenuBase {
             focusStateEnabled: this.option('focusStateEnabled'),
             animation: this.option('animation'),
             innerOverlay: true,
-            closeOnOutsideClick: this._closeOnOutsideClickHandler.bind(this),
+            hideOnOutsideClick: this._hideOnOutsideClickHandler.bind(this),
```
Comment (EugeniyKiyashko) [inline js/ui/context_menu/ui.context_menu.js:485]: hideOnOutsideClick: (e) => this._shouldHideOnOutsideClick(e)

**Review by Zedwag — DISMISSED** [review]

**Review by AryamnovEugeniy — DISMISSED** [review]

**Review by IgnatovDan — DISMISSED** [review]
ContextMenu changes looks ok

**Review by EugeniyKiyashko — DISMISSED** [review]

**Review by groshenkovamarina — DISMISSED** [review]

File: js/ui/context_menu.d.ts [hunk 2]
```diff
@@ -97,6 +98,14 @@ export interface dxContextMenuOptions<
      * @public
      */
     dataSource?: DataSourceLike<Item, TKey>;
+    /**
+     * @docid
+     * @default true
+     * @type_function_param1 event:event
```
Comment (IlyaKhD) [inline js/ui/context_menu.d.ts:105]: Add `@type boolean | function`

File: js/ui/overlay.d.ts [hunk 3]
```diff
@@ -89,6 +90,20 @@ export interface dxOverlayOptions<TComponent> extends WidgetOptions<TComponent>
      * @public
      */
     elementAttr?: any;
+    /**
+     * @docid
+     * @default false
+     * @type_function_param1 event:event
```
Comment (IlyaKhD) [inline js/ui/overlay.d.ts:97]: Add `@type boolean | function`

File: js/ui/popover.d.ts [hunk 4]
```diff
@@ -98,6 +99,20 @@ export interface dxPopoverOptions<TComponent> extends dxPopupOptions<TComponent>
        */
       name?: string;
     } | string;
+    /**
+     * @docid
+     * @type_function_param1 event:event
```
Comment (IlyaKhD) [inline js/ui/popover.d.ts:105]: Add `@type boolean | function`

File: js/ui/toast.d.ts [hunk 5]
```diff
@@ -83,6 +84,14 @@ export interface dxToastOptions extends dxOverlayOptions<dxToast> {
      * @public
      */
     displayTime?: number;
+    /**
+     * @docid
+     * @type_function_param1 event:event
```
Comment (IlyaKhD) [inline js/ui/toast.d.ts:90]: Add `@type boolean | function`

**Review by IlyaKhD — APPROVED** [review]

**Review by DokaRus — APPROVED** [review]

**Pull request merged**